packages/query/src/block-processor.ts

```diff
@@ -433,15 +434,7 @@ export class BlockProcessor implements BlockProcessorInterface {
         }),
       ]);
     } else if (action.case === 'actionDutchAuctionEnd' && action.value.auctionId) {
-      // Always a sequence number of 1 when ending a Dutch auction
-      const seqNum = 1n;
-
-      const metadata = getAuctionNftMetadata(action.value.auctionId, seqNum);
-
-      await Promise.all([
-        this.indexedDb.saveAssetsMetadata(metadata),
-        this.indexedDb.upsertAuction(action.value.auctionId, { seqNum }),
-      ]);
+      await processActionDutchAuctionEnd(action.value, this.querier.auction, this.indexedDb);
     } else if (action.case === 'actionDutchAuctionWithdraw') {
```

Comment:
I think helper functions for decluttering the block processor is great. For continuity, can we make `processActionDutchAuctionSchedule` and `processActionDutchAuctionWithdraw` helpers as well?

Reply:
Good call — done!